File: src/main/java/fr/sunderia/bomberman/Cooldown.java
package fr.sunderia.bomberman;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class Cooldown {
    
    private static final Map<String, Cooldown> cooldowns = new HashMap<>();
    private long start;
    private final int timeInSeconds;
    private final UUID id;
    private final String cooldownName;

    /**
     *
     * @param id Player's UUID {@link org.bukkit.entity.Player#getUniqueId()}
     * @param cooldownName Name of the cooldown
     * @param timeInSeconds Time in seconds
     */
    public Cooldown(UUID id, String cooldownName, int timeInSeconds) {
        this.id = id;
        this.cooldownName = cooldownName;
        this.timeInSeconds = timeInSeconds;
    }

    /**
     * @param id Player's UUID {@link org.bukkit.entity.Player#getUniqueId()}
     * @param cooldownName Name of the cooldown
     * @return True if the cooldown is running
     */
    public static boolean isInCooldown(UUID id, String cooldownName) {
        if(getTimeLeft(id, cooldownName) >= 1) {
            return true;
        } else {
            stop(id, cooldownName);
            return false;
        }
    }

    /**
     * Stop the cooldown
     * @param id Player's UUID {@link org.bukkit.entity.Player#getUniqueId()}
     * @param cooldownName Name of the cooldown
     */
    private static void stop(UUID id, String cooldownName) {
        Cooldown.cooldowns.remove(id + cooldownName);
    }

    /**
     * @param id The player's UUID {@link org.bukkit.entity.Player#getUniqueId()}
     * @param cooldownName The name of the cooldown
     * @return An instance of {@link Cooldown} from {@link Cooldown#cooldowns}
     * @throws NullPointerException if the cooldown doesn't exist
     */
    private static Cooldown getCooldown(UUID id, String cooldownName) {
        return cooldowns.get(id + cooldownName);
    }

    /**
     * @param id Player's UUID {@link org.bukkit.entity.Player#getUniqueId()}
     * @param cooldownName Name of the cooldown
     * @return Time left in seconds
     */
    public static int getTimeLeft(UUID id, String cooldownName) {
        Cooldown cooldown = getCooldown(id, cooldownName);
        int f = -1;
        if(cooldown != null) {
            long now = System.currentTimeMillis();
            int r = (int) (now - cooldown.start) / 1000;
            f = (r - cooldown.timeInSeconds) * (-1);
        }
        return f;
    }

    /**
     * Start the cooldown
     */
    public void start() {
        this.start = System.currentTimeMillis();
        cooldowns.put(this.id.toString() + this.cooldownName, this);
    }
}

File: src/main/kotlin/fr/sunderia/bomberman/utils/Cooldown.kt
package fr.sunderia.bomberman.utils

import java.util.*

/**
 *
 * @param id Player's UUID [net.minestom.server.entity.Entity.getUuid]
 * @param cooldownName Name of the cooldown
 * @param timeInSeconds Time in seconds
 */
class Cooldown(private val id: UUID, private val cooldownName: String, private val timeInSeconds: Int) {
    private var start: Long = 0

    /**
     * Start the cooldown
     */
    fun start() {
        start = System.currentTimeMillis()
        cooldowns[id.toString() + cooldownName] = this
    }

    companion object {
        private val cooldowns: MutableMap<String, Cooldown> = HashMap()

        /**
         * @param id Player's UUID [net.minestom.server.entity.Entity.getUuid]
         * @param cooldownName Name of the cooldown
         * @return True if the cooldown is running
         */
        fun isInCooldown(id: UUID, cooldownName: String): Boolean {
            return if (getTimeLeft(id, cooldownName) >= 1) {
                true
            } else {
                stop(id, cooldownName)
                false
            }
        }

        /**
         * Stop the cooldown
         * @param id Player's UUID [net.minestom.server.entity.Entity.getUuid]
         * @param cooldownName Name of the cooldown
         */
        private fun stop(id: UUID, cooldownName: String) {
            cooldowns.remove(id.toString() + cooldownName)
        }

        /**
         * @param id The player's UUID [net.minestom.server.entity.Entity.getUuid]
         * @param cooldownName The name of the cooldown
         * @return An instance of [Cooldown] from [Cooldown.cooldowns]
         * @throws NullPointerException if the cooldown doesn't exist
         */
        private fun getCooldown(id: UUID, cooldownName: String): Cooldown? {
            return cooldowns[id.toString() + cooldownName]
        }

        /**
         * @param id Player's UUID [net.minestom.server.entity.Entity.getUuid]
         * @param cooldownName Name of the cooldown
         * @return Time left in seconds
         */
        fun getTimeLeft(id: UUID, cooldownName: String): Int {
            val cooldown = getCooldown(id, cooldownName)
            var f = -1
            if (cooldown != null) {
                val now = System.currentTimeMillis()
                val r = (now - cooldown.start).toInt() / 1000
                f = (r - cooldown.timeInSeconds) * -1
            }
            return f
        }
    }
}